### PR TITLE
Issue 425/form error handling

### LINKED ISSUE #425: Forms should handle when there is an error making a request
### Description
When submitting forms, if the server fails on the request there is no indication of what went wrong. We need to handle these errors in order to show the user that their data was not saved.

This task would benefit from testing, so should be done once we have a way to handle tests.

### Requirements
- [ ] When submitting a form, if there is an error making the `POST` or `PATCH` request
  - [ ] Show a red `Alert` component at the top of the form informing the user there was an error submitting the data. 
  - [ ]  When the user submits again, hide the error, only showing it again if there is an error on that next request
  - [ ] Don't close the dialog for the form until the requset has successfully resolved
- [ ] Implement this on every form

### PR DESCRIPTION
**Pull request by djbusstop**
Resolves #425 

**General changes:**
- Move all forms in to `components/forms` (not sure if best idea, just flattens out the component folder heirarchy.)
  -  In process, renames forms to use `<Model><Action>Form` format
- Changes to CI pipleline
  - remove unused action
  - set max workers when running on CI
- [x] All forms submit button should only be enabled if no errors
- [x] Submit and Cancel buttongroup component

**Forms:**
- `TaskDetailsForm`
  - [x] `postTask` throws error if res is not an OK status code
  - [x] `patchTask` throws error if res is not an OK status code
  -  `createTask` action from speed dial
    - [x] Tests successfully posting Offline Task
    - [x] Tests that error alert shows when posting task returns error from server
  - Edit Task feature from action buttons
    - [x] Tests successfully editing existing task
    - [x] Tests that error alert shows when attempting to edit and error from server 
 
- `CampaignDetailsForm` 
  - [x] Simplify abstraction in campaign details form
  - [x] `postCampaign` throws error if res is not an OK status code 
  - [x] `patchCampaign` throws error if res is not an OK status code
  - `createCampaign` action from speed dial  
    - [x] Tests successfully posting campaign
    - [x] Tests that error alert shows
  - Edit campaign feature from action buttons
    - [x] Tests successfully editing existing campaign
    - [x] Tests that error alert shows when attempting to edit and error from server

### CODE REVIEW

**Comment by djbusstop** [comment]
Forms to change:

- Create Campaign
- Edit Campagaign
- Delete Campaign

- Create Task
- Edit Task
- Delete Task
- Publish Task

File: src/components/organize/campaigns/CampaignActionButtons.tsx [hunk 1]
```diff
@@ -1,16 +1,17 @@
 /* eslint-disable react-hooks/exhaustive-deps */
+import { Alert } from '@material-ui/lab';
 import Link from 'next/link';
 import { useRouter } from 'next/router';
 import { Box, Button, Menu, MenuItem } from '@material-ui/core';
 import { Delete, Settings } from '@material-ui/icons';
-import { FormattedMessage as Msg, useIntl } from 'react-intl';
+import { FormattedMessage, FormattedMessage as Msg, useIntl } from 'react-intl';
```
Comment (richardolsson) [inline src/components/organize/campaigns/CampaignActionButtons.tsx:7]: This is importing `FormattedMessage` twice.

File: src/components/organize/tasks/TaskActionButtons/index.tsx [hunk 2]
```diff
@@ -1,8 +1,9 @@
 /* eslint-disable react-hooks/exhaustive-deps */
+import { Alert } from '@material-ui/lab';
 import { useRouter } from 'next/router';
 import { Box, Button, Menu, MenuItem } from '@material-ui/core';
 import { Delete, Settings } from '@material-ui/icons';
-import { FormattedMessage as Msg, useIntl } from 'react-intl';
+import { FormattedMessage, FormattedMessage as Msg, useIntl } from 'react-intl';
```
Comment (richardolsson) [inline src/components/organize/tasks/TaskActionButtons/index.tsx:6]: `FormattedMessage` is being imported twice.

**Review by richardolsson — CHANGES_REQUESTED** [review]
Awesome work on this! I found some minor mistakes (I think?) and had one idea on how to improve the tests that submit forms, but other than that this is spotless.

File: playwright/tests/organize/campaign-detail/action-buttons.spec.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
+import { expect } from '@playwright/test';
+import test from '../../../fixtures/next';
+
+import KPD from '../../../mockData/orgs/KPD';
+import ReferendumSignatures from '../../../mockData/orgs/KPD/campaigns/ReferendumSignatures';
+
+test.describe.only('Campaign action buttons', async () => {
```
[the remark this answers is not in the document]
Comment (djbusstop) [reply playwright/tests/organize/campaign-detail/action-buttons.spec.ts:7]: nope!

**Review by richardolsson — APPROVED** [review]
LGTM! Ready to merge!

**Pull request merged**